index.ts

```diff
@@ -0,0 +1,552 @@
+  nonsky?: boolean; // Prevented from being executed or selected in a Sky Battle.
+  powder?: boolean; // Has no effect on Pokemon which are Grass-type, have the Ability Overcoat, or hold Safety Goggles.
+  protect?: boolean; // Blocked by Detect, Protect, Spiky Shield, and if not a Status move, King's Shield.
+  pulse?: boolean; // Power is multiplied by 1.5 when used by a Pokemon with the Ability Mega Launcher.
+  punch?: boolean; // Power is multiplied by 1.2 when used by a Pokemon with the Ability Iron Fist.
+  recharge?: boolean; // If this move is successful, the user must recharge on the following turn and cannot make a move.
+  reflectable?: boolean; // Bounced back to the original user by Magic Coat or the Ability Magic Bounce.
+  snatch?: boolean; // Can be stolen from the original user and instead used by another Pokemon using Snatch.
+  sound?: boolean; // Has no effect on Pokemon with the Ability Soundproof.
+}
+
+type MoveCategory: 'Physical' | 'Special' | 'Status';
+
+type MoveTarget =
+  // single-target
+  'normal'|'any'|'adjacentAlly'|'adjacentFoe'|'adjacentAllyOrSelf'|
+  // single-target, automatic
+  'self'|'randomNormal'|
+  // spread
+  'allAdjacent'|'allAdjacentFoes'|
+  // side and field
+  'allySide'|'foeSide'|'all';
+
+interface SelfEffect {
+  volatileStatus?: ID;
+  boosts?: Partial<BoostsTable>;
+}
+
+interface SecondaryEffect {
+  chance?: number;
+  boosts?: Partial<BoostsTable>;
+  self?: SelfEffect;
+  status?: StatusName;
+  volatileStatus?: ID;
+}
+
+type ContestType = 'Cool' | 'Beautiful' | 'Cute' | 'Clever' | 'Tough';
+
+interface CommonMove<TypeT, ItemT, MoveT> {
+  accuracy: number | true;
+  basePower: number;
+  flags: MoveFlags;
+  pp: number;
+  target: MoveTarget;
+  type: TypeT;
+
+  alwaysCrit?: boolean;
+  alwaysHit?: boolean;
+  basePowerModifier?: number;
+  boosts?: Partial<BoostsTable> | false
+  bypassesProtect?: boolean;
+  category: MoveCategory;
+  contestType?: ContestType;
+  critModifier?: number;
+  critRatio?: number;
+  damage?: number | 'level';
+  defensiveCategory?: MoveCategory;
+  drain?: [number, number]; // fraction
+  forceSTAB?: boolean;
+  forceSwitch?: boolean;
+  givesHealth?: boolean;
+  hasCustomRecoil?: boolean;
+  hasPriority?: boolean;
+  hasSecondaryEffect?: boolean;
+  heal?: [number, number]; // fraction
+  ignoreAbility?: boolean;
+  ignoreAccuracy?: boolean;
+  ignoreDefensive?: boolean;
+  ignoreEvasion?: boolean;
+  ignoreImmunity?: boolean | TypeT;
+  ignoreNegativeOffensive?: boolean;
+  ignoreOffensive?: boolean;
+  ignorePositiveDefensive?: boolean;
+  ignorePositiveEvasion?: boolean;
+  ignoresBurn?: boolean;
+  ignoresDefenseBoosts?: boolean;
+  infiltrates?: boolean;
+  isFutureMove?: boolean;
+  isSpread?: boolean | 'allAdjacent';
+  isZ?: ItemT;
+  multiaccuracy?: boolean;
+  multihit?: number | [number, number]; // range
+  noCopy?: boolean;
+  noDamageVariance?: boolean;
+  noFaint?: boolean;
+  noMetronome?: MoveT[];
```

Comment:
this is a field that is only set on `Metronome`. change to a flag and move to the moves it applies to